scalameter-core/src/main/scala/org/scalameter/execution/invocation/instrumentation/Instrumentation.scala

```diff
@@ -0,0 +1,107 @@
+package org.scalameter.execution.invocation.instrumentation
+
+import java.io._
+import java.util.jar.{JarEntry, JarOutputStream, Attributes}
+import java.util.zip.ZipFile
+import org.objectweb.asm.{ClassReader, ClassWriter}
+import scala.collection.convert.decorateAsScala._
+import org.scalameter._
+import org.scalameter.execution.invocation._
+
+
+private[scalameter] object Instrumentation {
+  /** Reads classes from given location and transforms them if they match a predicate.
+   *
+   *  Note that it supports reading from directories and jars.
+   *
+   *  @param in location of classes to read from
+   *  @param p matches if class should be instrumented
+   *  @param instrumentFrom instruments class from a [[java.io.InputStream]]
+   *  @return [[scala.Iterator]] of tuples in a form of (fullClassName, instrumentedClassBytes)
+   */
+  private def transformClasses(in: File)(p: String => Boolean)(instrumentFrom: (String, InputStream) => Array[Byte]): Iterator[(String, Array[Byte])] = {
+    val dirFilter = new FilenameFilter {
+      def accept(file: File, s: String): Boolean = new File(file, s).isDirectory
+    }
+
+    val classFilter = new FilenameFilter {
+      def accept(file: File, s: String): Boolean = s.endsWith(".class")
+    }
+
+    def _transformClasses(in: File, rootPath: String): Iterator[(String, Array[Byte])] = {
```

Comment:
As far as I remember there is also an underscore in the two other methods from the previous PRs:
`def _unpickle(times: Int, from: Int)` in the `Pickler.unpickle` and
`def _readBytes()` in the `IO.readFromInputStream`
Should they be also changed? - I can do it in the next PR after we iterate over this.